Arrowgene.Ddon.Shared/Entity/Structure/CDataJobOrbDevoteElement.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Arrowgene.Buffers;
+using Arrowgene.Ddon.Shared.Model;
+
+namespace Arrowgene.Ddon.Shared.Entity.Structure
+{
+    public class CDataJobOrbDevoteElement
+    {
+        public CDataJobOrbDevoteElement()
+        {
+            RequiredElementIDList = new List<CDataCommonU32>();
+            RequiredQuestList = new List<CDataCommonU32>();
+            Unk0 = 0;
+        }
+
+        public UInt32 ElementId {  get; set; }
+        public JobId JobId { get; set; }
+        public byte Unk0 { get; set; }
+        public UInt32 RequireOrb { get; set; }
+        public byte OrbRewardType { get; set; }
+        public bool IsReleased { get; set; }
+        public UInt32 ParamId { get; set; }
+        public UInt32 ParamValue { get; set; }
+        public byte Unk1 { get; set; }
+        public UInt32 PosX { get; set; }
+        public UInt32 PosY { get; set; }
+        public bool DrawConnection { get; set; }
+
+        public List<CDataCommonU32> RequiredElementIDList { get; set; }
+        public List<CDataCommonU32> RequiredQuestList {  get; set; } 
+
+        public class Serializer : EntitySerializer<CDataJobOrbDevoteElement>
+        {
+            public override void Write(IBuffer buffer, CDataJobOrbDevoteElement obj)
+            {
+                WriteUInt32(buffer, obj.ElementId);
+                WriteByte(buffer, (byte)obj.JobId);
+                WriteByte(buffer, obj.Unk0);           // No idea what this does
+                WriteUInt32(buffer, obj.RequireOrb);   // amount of BO required for the upgrade
+                WriteByte(buffer, obj.OrbRewardType);  // Changes text between multiple effects in the UI
+                WriteUInt32(buffer, obj.ParamId);      // No idea what this does
+                WriteUInt32(buffer, obj.ParamValue);   // Shows next to OrbRewardType value
+                WriteByte(buffer, 0xff);               // Doesn't impact Position
+                WriteByte(buffer, 0xff);               // Doesn't Impact Position
+                WriteByte(buffer, 0xff);               // Doesn't impact Position
+                WriteByte(buffer, (byte) obj.PosX);    // This byte controls the x position
```

Comment:
Couldn't it happen to be a negative signed 32 bit integer?

Reply:
From what I saw the X would move from 1 to n, where N could be off screen.  The value was not a pixel amount either, but some fixed amounts from the left side of the screen. There doesn't appear to be a Y axis value as the debug data suggests for season 2. Further seen by the fact this bool which I marked as "draw connection" is in the place where the Y should be. Or at least the Y value doesn't seem to simply be a u32. Based on the fact that the X value reasonably could be a value of 1-7, I suppose that a byte for each coordinate would be sufficient for this UI coordinate system,

Reply:
What about a float? 🤔